From: olh@suse.de
Subject: compile fix for kernel-xen

	commit 985a34bd75cc8c96e43f00dcdda7c3fdb51a3026
	Author: Thomas Gleixner <tglx@linutronix.de>
	Date:   Sun Mar 9 13:14:37 2008 +0100

	x86: remove quicklists

	quicklists cause a serious memory leak on 32-bit x86,
	as documented at:

	http://bugzilla.kernel.org/show_bug.cgi?id=9991

	the reason is that the quicklist pool is a special-purpose
	cache that grows out of proportion. It is not accounted for
	anywhere and users have no way to even realize that it's
	the quicklists that are causing RAM usage spikes. It was
	supposed to be a relatively small pool, but as demonstrated
	by KOSAKI Motohiro, they can grow as large as:

	Quicklists:    1194304 kB

	given how much trouble this code has caused historically,
	and given that Andrew objected to its introduction on x86
	(years ago), the best option at this point is to remove them.

	[ any performance benefits of caching constructed pgds should
	be implemented in a more generic way (possibly within the page
	allocator), while still allowing constructed pages to be
	allocated by other workloads. ]

	Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
	Signed-off-by: Ingo Molnar <mingo@elte.hu>

---
 arch/x86/Kconfig |    5 +++++
 1 file changed, 5 insertions(+)

--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -253,6 +253,11 @@ config X86_XEN
 	  Choose this option if you plan to run this kernel on top of the
 	  Xen Hypervisor.
 
+config QUICKLIST
+	bool
+	depends on X86_XEN
+	default y
+
 config X86_ELAN
 	bool "AMD Elan"
 	depends on X86_32
